index.html

```diff
@@ -1213,59 +1216,94 @@
       <h2>
         Examples
       </h2>
-      <h3>
-        Playback of depth and color streams from same device group.
-      </h3>
-      <pre class="example">
-navigator.mediaDevices.getUserMedia({
-  video: {videoKind: {exact: "color"}, groupId: {exact: id}}
-}).then(function (stream) {
-    // Wire the media stream into a &lt;video&gt; element for playback.
-    // The RGB video is rendered.
-    var video = document.querySelector('#video');
-    video.srcObject = stream;
-    video.play();
+      <aside title="Playback of depth and color streams from same device group"
+      class="example">
+        <pre class="highlight">
+async function attachVideoStream(el, kind, groupId) {
+  const constraints = {
+    video: {
+      mandatory: {
+        groupId: groupId
+      },
+      videoKind: { exact: kind }
+    }
   }
-);
 
-navigator.mediaDevices.getUserMedia({
-  video: {videoKind: {exact: "depth"}, groupId: {exact: id}}
-}).then(function (stream) {
-    // Wire the depth-only stream into another &lt;video&gt; element for playback.
-    // The depth information is rendered in its grayscale representation.
-    var depthVideo = document.querySelector('#depthVideo');
-    depthVideo.srcObject = stream;
-    depthVideo.play();
-  }
-);
+  const stream = await navigator.mediaDevices.getUserMedia(constraints);
+
+  el.srcObject = stream;
+  el.play();
+
+  return el;
+}
+
+async function play() {
+  const colorEl = document.createElement("video");
+  const depthEl = document.createElement("video");
+
+  const body = document.querySelector('body');
+  body.appendChild(colorEl);
+  body.appendChild(depthEl);
+
+  // Assume that all our video inputs are depth stream capable.
+  const devices = await navigator.mediaDevices.enumerateDevices();
+  const sources = devices.filter(device =&gt; device.kind == "videoinput");
+
+  // Attached cameras usually comes last.
+  const camera = sources.pop();
+
+  // Regular RGB video will be rendered.
+  attachVideoStream(colorEl, "color", camera.groupId);
+
+  // Depth information will be rendered in its grayscale representation.
+  attachVideoStream(depthEl, "depth", camera.groupId);
+}
+
+play();
+
 </pre>
-      <h3>
-        WebGL Fragment Shader based post-processing
-      </h3>
-      <pre class="example">
-// This code sets up a video element from a depth stream, uploads it to a WebGL
-// texture, and samples that texture in the fragment shader, reconstructing the
-// 16-bit depth values from the red and green channels.
-navigator.mediaDevices.getUserMedia({
-  video: {videoKind: {exact: "depth"}}
-}).then(function (stream) {
-  // wire the stream into a &lt;video&gt; element for playback
-  var depthVideo = document.querySelector('#depthVideo');
-  depthVideo.srcObject = stream;
-  depthVideo.play();
-}).catch(function (reason) {
-  // handle gUM error here
-});
+      </aside>
```

Comment:
This approach is going to be deleted soon - reconstruction from split components is not supported.
We need to base the example on floats - like in https://software.intel.com/en-us/blogs/2017/02/10/depth-camera-capture-in-html5.